WebRoot/js/zimbraMail/calendar/view/ZmCalColView.js

```diff
@@ -2842,11 +2842,11 @@ function(ev) {
         var col = data.view._getColFromX(data.gridX);
 		Dwt.setVisible(data.newApptDivEl, false);
 		if (data.isAllDay) {
-			appCtxt.getCurrentController().newAllDayAppointmentHelper(data.startDate, data.endDate, null, mouseEv.shiftKey);
+			appCtxt.getCurrentController().newAllDayAppointmentHelper(data.startDate, data.endDate, col.cal.id, mouseEv.shiftKey);
```

Comment:
It causes an error when the view is Work Week, Week and Month views (i.e. not Day view) because `col.cal` is `undefined`. `data.view._getColFromX(data.gridX)` at line 2842 may return `null` for some reason. Need null/undefined check on both `col` and `col.cal`.
Line 2849 as well.

When Merged view in Day view, "Calendar" seems to always be selected. Is it expected design?

Reply:
As suggested, a null/undefined check for col.cal has been added to the code. 
I have looked into merged view for Day view in the current zimbra dev environment and the "Calendar" dropdown is always selected when creating new appointments and new all day appointments. For existing appointments, the context menu points to the specific calendar instance. It's not impacted by my code change. In merged view, there doesn't seem to be a way to preselect a specific calendar hence default calendar is selected. I have confirmed this with Bhavin as well.